Add blogs

eleensmathew opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Add Blogs

<!--- Describe your changes in detail -->

Added functionality to add blogs and fuxed the templates for seeing all blogs and seeing blog detail. Also implemented nested comments.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

Fixes # (https://github.com/ContriHUB/Youdemy/issues/2)

## Type of change

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

eleensmathew commented:
![127 0 0 1_8000_blog_single_8_](https://github.com/ContriHUB/Youdemy/assets/92302620/20c97e8e-1cdf-4d33-b9a1-36d2752a4067)
blog_detail currently looks like this

eleensmathew commented:
Have not made changes to Category, Popular articles and Archives

kunal2812 commented on main/urls.py:
```diff
@@ -20,6 +20,7 @@
     path('add_titles/<int:chapter_id>/',views.create_title_model_form,name='create_title_model_form'),
     path('add_question/<int:chapter_id>/',views.create_question,name='create_question'),
     path('add_chapter/<int:course_id>/',views.create_chapter,name='create_chapter'),
+    path('add_blog/',views.create_blog,name='create_blog'),
```
Where is the option on frontend to create blog ? Can't see it anywhere

eleensmathew replied:
I'll work on it

kunal2812 replied:
Also, ensure that I am able to replicate the blog as closely as possible to to static one using the functionality you implemented.

eleensmathew commented:
I added link in the all blogs pags 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/a35716b8-67ac-4ab3-b86c-4c27c14a4be4)
Added edit and delete blogs 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/ad0c3519-e37e-4fa9-8167-d478658d4d4b)
Slightly improved display of like and dislike button 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/ec91d1f5-89f7-4121-adb9-294e71fc9bbd)
Changed display of email in comments to first name 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/3602cd1d-754f-44d3-8ffb-9343c5278753)

eleensmathew commented:
I have made all the suggested changes. If you have any comments please let me know

kunal2812 commented:
After the blog is created when I click the blog card to view full blog I'm getting this error.
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/e8b618fc-e148-425d-9e71-6fdc5e7a61cd)

eleensmathew commented:
That error is happening becaus the person is supposed to create a profile for himself. Right now the userprofile contains the bio so do you just want me to put some default value for all the fields of userprofile?

kunal2812 commented:
Yes those fields which are not asked during the signup can have the default values for now.

eleensmathew commented:
Resolved the error. You have to make migrations and try creating a user from scratch to test it. 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/8978da85-beea-4028-a7eb-bcbd3b94311e)
The default values are seen above.

kunal2812 commented:
Issues:
1. If the user is not signed in don't show the functionality to add blog. Leads to this error.
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/4a6e87e5-b76c-4c9f-9bcc-a7adfa033bc1)

2. Have you added paging to the blogs or are they appended in the same page itself ? If you are showing all blogs in same page then remove this from frontend.
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/fa6574cb-c4c2-4f08-8d25-d0960c49707c)

3. All blog cards should be of same size as shown in #2. The blog heading should be shown in 2 lines and preview of blog content in 4 lines. Keep this consistent across all cards.
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/b287caea-57fb-4f64-bcd3-0de9dbad2f97)

4. Fix css of blog details page footer
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/8c4dbcfd-2614-4456-ac92-3c827aa53da5)

5. Currently blog content looks a bit congested. Break blog content into 3 parts - Introduction, Main Body, Conclusion and Place the blog thumbnail after the Introduction (refer static page for more details). 

6. If I comment via comment form then I am on the same page but when I comment on the thread then I'm taken back to Blog cards page (Same page should refresh).

7. Like Dislike not working
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/3e285545-9094-497c-830a-2d8d123bfa13)

8. This takes me to "Home" page.
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/361ac270-6ece-468d-b4e8-492588fafcdc)

9. Time is in UTC. Change it to local time.
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/6259eecd-a203-4026-88a5-0740fa97b76c)

eleensmathew commented:
changed the timezone 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/84c11e7e-4544-4842-aacb-b035caffb7f1)
added intro and conclusion 
![127 0 0 1_8000_blog_1_](https://github.com/ContriHUB/Youdemy/assets/92302620/d45b5285-f5ce-406b-a87b-17e8330c7b5c)
modified card display 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/92b6cdd5-c3be-450e-9a29-1de9594995db)
changed all the incorrect links and made add blog functionality only to registered users. Fixed like and dislike. Removed next page option as well.
However the css of the footer in consistent for me on all pages, so can you try clearing the cache and letting me know what happens?

kunal2812 reviewed (changes requested):
> 3. The blog heading should be shown in 2 lines and preview of blog content in 4 lines. Keep this consistent across all cards.
>    ![image](https://user-images.githubusercontent.com/59749539/278898651-b287caea-57fb-4f64-bcd3-0de9dbad2f97.png)

This is not done yet.

Fix typo here
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/64067319-b470-466c-916a-7fb90adf0c52)

![image](https://github.com/ContriHUB/Youdemy/assets/59749539/86f4f03c-3577-4134-9e06-e37ba3915a0d)
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/febe35f4-fad6-4da0-b386-2c0a2df76ac0)

Check blogs page and blogs detail page. There is this extra white space.

Getting this error when I enter 0 tags when I create blog. And even though I was getting the error the blog was anyway still created. Either add the functionality to not add any tags or force the user to add atleast 1 tag (First option is preferable).
![image](https://github.com/ContriHUB/Youdemy/assets/59749539/8f40b3c8-dfc2-4c88-b9b1-172fb7ad0185)

eleensmathew commented:
do these cards look ok?
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/b0875269-6345-459a-b3c6-45fba7b7914d)
i made the cards all uniform height

kunal2812 commented:
> do these cards look ok? ![image](https://user-images.githubusercontent.com/92302620/279019463-b0875269-6345-459a-b3c6-45fba7b7914d.png) i made the cards all uniform height

The blog card should consist of three sections: the Blog Heading, Blog Content, and the Read More part. Each of these sections should have a fixed height within the card, and they should all be aligned on the same line.

eleensmathew commented:
i hope this is fine now ?
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/27d44e20-d6f2-4772-b5aa-1593f78b5f24)

kunal2812 commented:
> i hope this is fine now ? ![image](https://user-images.githubusercontent.com/92302620/279028309-27d44e20-d6f2-4772-b5aa-1593f78b5f24.png)

Yes. But keep the First name of user and no. Of comments in line with "Read More" like before

eleensmathew commented:
fixed the cards
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/675e65dc-8c04-4362-a44a-2e9ea2e6142b)

fixed footer and header white space 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/5d0b61b0-d794-41a3-9c47-67ec61ba426a)
and fixed empty tags problem

kunal2812 commented:
When an unregistered user tries to like, dislike, or comment, they should be redirected to the sign-up page, as this action would otherwise result in an error.

Everything else appears to be okay.

eleensmathew commented:
Can I display something like to view comments sign in? and to like and dislike this post sign in and i attach the link to home page if they click sign in?

eleensmathew commented:
Something like this 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/fc773ea0-ef13-4044-91c0-c2b00e4af2e3)

kunal2812 commented:
> Can I display something like to view comments sign in? and to like and dislike this post sign in and i attach the link to home page if they click sign in?

Yes you can do that.

eleensmathew commented:
Made all the required changes. look like this now 
![image](https://github.com/ContriHUB/Youdemy/assets/92302620/fc6258ff-442f-482e-a1e6-54eedf1b9909)

kunal2812 reviewed (approved):
LGTM
